components/ledger/pkg/api/controllers/transaction_controller.go

```diff
@@ -222,15 +224,17 @@ func (ctl *TransactionController) PostTransaction(w http.ResponseWriter, r *http
 		Metadata:  payload.Metadata,
 	}
 
-	res, logs, err := l.CreateTransaction(r.Context(), preview, script)
+	res, logHandler, err := l.CreateTransaction(r.Context(), preview, script)
 	if err != nil {
 		apierrors.ResponseError(w, r, err)
 		return
 	}
 
-	if err := logs.Wait(r.Context()); err != nil {
-		apierrors.ResponseError(w, r, err)
-		return
+	if !preview {
```

Comment:
Normally it's automatically handled by the fact that the log handler that is returned is empty !

Reply:
Alright